api/src/services/observation-service.ts

```diff
@@ -1019,4 +1194,30 @@ export class ObservationService extends DBService {
     // Return true if both environments and measurements are valid
     return true;
   }
+
+  /**
+   * Gets the code id value with a matching name from a set of options. If the function returns null, the
+   * request should probably throw an error.
+   *
+   * @param cellValue The name of a code to find the id for
+   * @param codeOptions The reference codes from which to find the id
+   * @param defaultValue A default value for when cellValue is null
+   * @returns
+   */
+  _getCodeIdFromCellValue(cellValue: string | null, codeOptions: ICode[], defaultValue?: string | null): number | null {
+    const value = cellValue?.toLowerCase(); // Normalize the cell value
+
+    // No value exists and no default value, so must be null
+    if (!value && !defaultValue) {
+      return null;
+    }
+
+    // The cell has no value so return the default
+    if (!value && defaultValue) {
+      return codeOptions.find((option) => option.name.toLowerCase() === defaultValue.toLowerCase())?.id || null;
```

Comment:
suggestion: We could pass the defaultCodeId as a param to prevent having to loop through all the codes for every code cell.
ie: Loop through once externally vs every time internally